management/client.go

```diff
@@ -26,32 +28,49 @@ func (e *AzureError) Error() string {
 
 // Client provides a client to the Azure API.
 type Client struct {
+	managementURL   string
 	publishSettings publishSettings
 }
 
-//NewAnonymouseClient creates a new azure.Client with no credentials set.
+// ClientConfig provides a configuration for use by a Client
+type ClientConfig struct {
+	ManagementURL string
+}
+
+// NewAnonymousClient creates a new azure.Client with no credentials set.
 func NewAnonymousClient() Client {
 	return Client{}
 }
 
-//NewClientFromPublishSettingsFile creates a new azure.Client and imports the publish
-//settings from the specified file path.
-func NewClientFromPublishSettingsFile(publishSettingsFilePath string) (Client, error) {
-	client := Client{}
-	err := client.importPublishSettingsFile(publishSettingsFilePath)
-	if err != nil {
-		return client, err
-	}
-	return client, nil
+// NewClient creates a new Client using the given subscription ID and
+// management certificate
+func NewClient(subscriptionID string, managementCert []byte) (Client, error) {
+	return makeClient(subscriptionID, managementCert, defaultAzureManagementURL)
```

Comment:
this should really call `NewClientFromConfig` with "default config" (in this case we just have the URL but in the future, default retry policy, etc).